packages/google_sign_in/lib/testing.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter/services.dart' show MethodCall;
+
+/// A fake backend that can be used to test components that require a valid
+/// [GoogleSignInAccount].
+///
+/// Example usage:
+///
+/// GoogleSignIn googleSignIn;
+/// FakeSignInBackend fakeSignInBackend;
+///
+/// setUp(() {
+///   googleSignIn = new GoogleSignIn();
+///   fakeSignInBackend = new FakeSignInBackend();
+///   fakeSignInBackend.user = new FakeUser(
+///     id: 123,
+///     email: 'jdoe@example.org',
+///   );
+///   googleSignIn.channel.setMockMethodCallHandler(
+///       fakeSignInBackend.handleMethodCall);
+/// });
+///
+class FakeSignInBackend {
+  FakeUser user;
```

Comment:
Add docs to explain that this does not represent the signed-in user, but instead represents the user that will be signed in when `signIn` or `signInSilently` is called in the plugin.